zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/RestoreFromBackupTool.java

```diff
@@ -0,0 +1,120 @@
+package org.apache.zookeeper.server.backup;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.zookeeper.common.ConfigException;
+import org.apache.zookeeper.server.backup.storage.BackupStorageProvider;
+import org.apache.zookeeper.server.backup.storage.impl.FileSystemBackupStorage;
+import org.apache.zookeeper.server.persistence.FileTxnSnapLog;
+
+/**
+ * TODO: This is not a complete implementation.
+ * RestoreFromBackupTool skeleton.
+ */
+public class RestoreFromBackupTool {
+  private static final int MAX_RETRIES = 10;
+
+  BackupStorageProvider storage;
+
+  FileTxnSnapLog snapLog;
+  long zxidToRestore;
+  boolean dryRun = false;
+
+  List<BackupFileInfo> logs;
+  List<BackupFileInfo> snaps;
+  List<BackupFileInfo> filesToCopy;
+
+  int mostRecentLogNeededIndex;
+  int snapNeededIndex;
+  int oldestLogNeededIndex;
+
+  public static void usage() {
+    System.out.println("Usage: RestoreFromBackupTool restore <restore_point> <backup_store> <data_destination> <log_destination>");
+    System.out.println("    restore_point: the point to restore to, either the string 'latest' or a zxid in hex format.");
+    System.out.println("    backup_store: the connection information for the backup store");
+    System.out.println("       For GPFS the format is: gpfs:<config_path>:<backup_path>:<namespace>");
+    System.out.println("           config_path: the path to the hdfs configuration");
+    System.out.println("           backup_path: the path within hdfs where the backups are stored");
+    System.out.println("    data_destination: local destination path for restored snapshots");
+    System.out.println("    log_destination: local destination path for restored txlogs");
+  }
+
+  /**
+   * Parse and validate arguments to the tool
+   * @param args the set of arguments
+   * @return true if the arguments parse correctly; false in all other cases.
+   * @throws IOException if the backup provider cannot be instantiated correctly.
+   */
+  public void parseArgs(String[] args) throws ConfigException {
+    // Check the num of args
+    if (args.length != 4 && args.length != 5) {
```

Comment:
From a cli perspective, it is not a good idea to fix the number of arguments. Also order of arguments should not matter. The better way could be scanning the input arguments to find whether there is a matching option.